File: DSA/SAMAGRA_590014891/Day_50_Question_1.c
#include <stdio.h>

int maxSumSubarray(int arr[], int n, int k) {
    if (n < k) return -1;

    int max_sum = 0, window_sum = 0;

    for (int i = 0; i < k; i++)
        window_sum += arr[i];

    max_sum = window_sum;

    for (int i = k; i < n; i++) {
        window_sum += arr[i] - arr[i - k];
        if (window_sum > max_sum)
            max_sum = window_sum;
    }

    return max_sum;
}

int main() {
    int arr[] = {100, 200, 300, 400};
    int k = 2;
    int n = sizeof(arr) / sizeof(arr[0]);
    printf("%d\n", maxSumSubarray(arr, n, k));
    return 0;
}

File: DSA/akashyadav_590014126/Day_50_Question_1.py
def max_sum_subarray(arr, k):
    if len(arr) < k:
        return None 

    window_sum = sum(arr[:k])
    max_sum = window_sum

    for i in range(k, len(arr)):
        window_sum += arr[i] - arr[i - k]
        max_sum = max(max_sum, window_sum)

    return max_sum

# Example usage
print(max_sum_subarray([300, 400, 500, 600], 3))